indexing-service/src/main/java/io/druid/indexing/overlord/autoscaling/TwoCloudWorkerProvisioningStrategy.java

```diff
@@ -0,0 +1,161 @@
+/*
+ * Licensed to Metamarkets Group Inc. (Metamarkets) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. Metamarkets licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.druid.indexing.overlord.autoscaling;
+
+import com.google.common.base.Predicate;
+import com.google.common.base.Supplier;
+import com.google.common.base.Suppliers;
+import com.google.inject.Inject;
+import com.metamx.common.concurrent.ScheduledExecutors;
+import io.druid.indexing.common.task.Task;
+import io.druid.indexing.overlord.TasksAndWorkers;
+import io.druid.indexing.overlord.WorkerTaskRunner;
+import io.druid.indexing.overlord.setup.TwoCloudConfig;
+import io.druid.indexing.overlord.setup.WorkerBehaviorConfig;
+
+import javax.annotation.Nullable;
+import java.util.concurrent.ScheduledExecutorService;
+
+import static io.druid.indexing.common.task.TaskLabels.getTaskLabel;
+
+public class TwoCloudWorkerProvisioningStrategy extends AbstractWorkerProvisioningStrategy
+{
+  public static final Supplier<ScheduledExecutorService> EXEC_FACTORY = new Supplier<ScheduledExecutorService>()
```

Comment:
Can this just be `Execs.scheduledSingleThreaded("TwoCloudWorkerProvisioningStrategy-manager--%d")`?